src/test/java/co/rsk/federate/signing/hsm/message/PowHSMSignerMessageTest.java

```diff
@@ -78,109 +138,187 @@ void equality() {
-                "cbe0517c17e1012102e87cd90f3cb0d64eeba797fbb8f8ceaadc09e0128afbaefb0ee9535875ea395400000000";
-        BtcTransaction btcTx = new BtcTransaction(RegTestParams.get(), Hex.decode(rawTx));
+        BtcTransaction btcTx = new BtcTransaction(RegTestParams.get(), Hex.decode(simpleRawBtcTx));
         int inputIndex = 0;
         TransactionReceipt txReceipt = new TransactionReceipt();
         List<Trie> receiptMerkleProof = new ArrayList<>();
         receiptMerkleProof.add(new Trie());
-        Sha256Hash sigHash = Sha256Hash.wrap("0000000000000000000000000000000000000000000000000000000000000001");
-        SignerMessage message = new PowHSMSignerMessage(btcTx, inputIndex, txReceipt, receiptMerkleProof, sigHash);
+        Sha256Hash sigHash = Sha256Hash.wrap(
+            "0000000000000000000000000000000000000000000000000000000000000001");
+        PowHSMSignerMessage message = new PowHSMSignerMessage(btcTx, inputIndex, txReceipt,
+            receiptMerkleProof, sigHash);
+
+        assertEquals(sigHash, message.getSigHash());
+    }
+
+    @ParameterizedTest
+    @MethodSource("getMessageToSignLegacyArgProvider")
+    void getMessageToSign_whenSighashLegacyMode_ok(int version, JsonNode expectedMessageToSend) {
+        // arrange
+        TransactionReceipt txReceipt = new TransactionReceipt();
+        List<Trie> receiptMerkleProof = new ArrayList<>();
+        receiptMerkleProof.add(new Trie());
+        Sha256Hash sigHash = BitcoinTestUtils.createHash(1);
+        PowHSMSignerMessage powHSMSignerMessage = new PowHSMSignerMessage(pegoutBtcTx,
+            FIRST_OUTPUT_INDEX, txReceipt, receiptMerkleProof, sigHash);
 
-        assertEquals(sigHash, ((PowHSMSignerMessage) message).getSigHash());
+        // act
+        JsonNode actualMessageToSign = powHSMSignerMessage.getMessageToSign(version);
+
+        // assertions
+        Assertions.assertEquals(expectedMessageToSend, actualMessageToSign);
+    }
+
+    @Test
+    void getMessageToSign_whenSighashSegwitMode_ok() {
+        // arrange
+        List<Coin> outpointValues = coinListOf(50_000_000, 75_000_000, 100_000_000);
+
+        BtcTransaction fundingTransaction = new BtcTransaction(btcMainnetParams);
+        fundingTransaction.addInput(
+            BitcoinTestUtils.createHash(1),
+            FIRST_OUTPUT_INDEX,
+            new Script(new byte[]{})
+        );
+
+        for (Coin outpointValue : outpointValues) {
+            fundingTransaction.addOutput(outpointValue, activeFederation.getAddress());
+        }
+
+        BtcTransaction segwitPegoutBtcTx = new BtcTransaction(btcMainnetParams);
+
+        TransactionWitness txWitness = new TransactionWitness(outpointValues.size());
+        for (int i = 0; i < outpointValues.size(); i++) {
+            Coin outpointValue = outpointValues.get(i);
+
+            segwitPegoutBtcTx.addInput(fundingTransaction.getOutput(i));
+            txWitness.setPush(i, new byte[]{0x1});
```

Comment:
This isn't very realistic, the witness should have the same data as the script sig of a legacy transaction. I think it's important we use realistic transactions to test this.

We can consider creating a BtcTransactionBuilder to help with this if there isn't something already existent